### PR TITLE
local test instance for gitlab

### PR DESCRIPTION
**Pull request by BeckerFrank**
for the issue #37 (gitlab Connector) we need a local test instance so I did the following

- add gitlab role
  this role has a special case for Mac with Apple Silicon (arm64)
- rework general setup
  add support for remote and local setups
- place the root certificate in a new cacerts so that we can use this in local runtime workspaces
- add artifactory role
  maybe needed in the future.

When we have finished this, I will post a first version of a gitlab test project

### CODE REVIEW

**Comment by ruspl-afed** [comment]
@BeckerFrank thank you so much for working on this!

This is not my area of expertise, so I hardly can provide valuable feedback here. Please feel free to merge if it works for you.

File: org.eclipse.mylyn.releng/multipass/roles/jenkins/files/jenkins-2.303.3/jenkins_home/config_disable.xml
Comment (gnl42) [inline org.eclipse.mylyn.releng/multipass/roles/jenkins/files/jenkins-2.303.3/jenkins_home/config_disable.xml:1]: We stopped supporting hudson last year. Do we still need this?

File: org.eclipse.mylyn.releng/multipass/mylyn_setup.yml [hunk 1]
```diff
@@ -7,90 +7,28 @@
...
-        - createValidCertificates
-        - base_software
+        - global
+    - name: debug distribution
+      debug:
+        msg: "{{ localhostPlatform }}"
+
 #
-# Alle normalen Knoten mit Zertifikaten versorgen
+# Alle normalen Knoten Docker starten
```
Comment (wimjongman) [inline org.eclipse.mylyn.releng/multipass/mylyn_setup.yml:17]: Deutsch ist gut fur Mich, aber ich denke das das nicht fur die Ganzen Welt gultet.

File: org.eclipse.mylyn.releng/multipass/roles/jenkins/files/update-center.crt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```
Comment (wimjongman) [inline org.eclipse.mylyn.releng/multipass/roles/jenkins/files/update-center.crt:1]: Is this intended?

**Review by wimjongman — COMMENTED** [review]
I don't know this magic. It looks like a lot of configuration.

File: org.eclipse.mylyn.releng/multipass/roles/jenkins/files/update-center.crt (hunk 2 above)
Comment (BeckerFrank) [reply org.eclipse.mylyn.releng/multipass/roles/jenkins/files/update-center.crt:1]: No not really. This was left from a test that you can use an updatesite without redirect to different mirrors.
Will try to delete the changes in the next days.

File: org.eclipse.mylyn.releng/multipass/mylyn_setup.yml (hunk 1 above)
Comment (BeckerFrank) [reply org.eclipse.mylyn.releng/multipass/mylyn_setup.yml:17]: I will fix this with the next commit.

File: org.eclipse.mylyn.releng/multipass/roles/jenkins/files/jenkins-2.303.3/jenkins_home/config_disable.xml
Comment (BeckerFrank) [reply org.eclipse.mylyn.releng/multipass/roles/jenkins/files/jenkins-2.303.3/jenkins_home/config_disable.xml:1]: Yes, but Jenkins is still supported.

Comment (gnl42) [reply org.eclipse.mylyn.releng/multipass/roles/jenkins/files/jenkins-2.303.3/jenkins_home/config_disable.xml:1]: Sigh, my bad. Forgot Jenkins "inherited" the hudson xml

Comment (BeckerFrank) [reply org.eclipse.mylyn.releng/multipass/roles/jenkins/files/jenkins-2.303.3/jenkins_home/config_disable.xml:1]: No problem, so I checked again that there were no more Hudson-specific parts and tidied up a bit.

**Pull request merged**